src/controller/UserController.php

```diff
@@ -157,14 +147,16 @@ public function disconnect()
     /**
      * render Template.
      *
-     * @param  string $templateName Template name to render
-     * @param  array $error error information to display
-     * @param  array $form content of the completed form 
-     * @param  array $session user session
-     * @return void
+     * @param string $templateName Template name to render
+     * @param array $error error information to display
+     * @param array $form content of the completed form
+     * @param array $session user session
+     * @throws LoaderError
+     * @throws RuntimeError
+     * @throws SyntaxError
      */
-    private function render(string $templateName, array $error, array $form, array $session)
+    private function render(string $templateName, array $errors = [], array $form = [], array $session = [])
     {
-        $this->twig->render($templateName, ['error' => $error, 'form' => $form, 'SESSION' => $session]);
+        echo $this->twig->render($templateName, ['error' => $errors, 'form' => $form, 'SESSION' => $session]);
```

Comment:
En fait ton problème de `echo` vient du fait que la fonction `render()` ne retournait rien. 

Si tu voulais faire un ` echo $this->render(...)` il fallait que tu spécifies à dans ta fonction un `return $this->twig->render(....)`

Reply:
Ok